src/states/elder/mod.rs

```diff
@@ -368,6 +368,14 @@ impl Elder {
         }
     }
 
+    fn finalise_prune(&mut self) -> Result<(), RoutingError> {
+        let _ = self.chain.finalise_prune();
+        self.chain.reset_candidate();
+        self.peer_mgr.reset_candidate();
+        self.init_parsec(); // We don't reset the chain on pruning.
```

Comment:
Prunning does not seem to carry over any events the way prefix change does, how does that work?

Reply:
It happens immediately when got the consensused `ParsecPrune` event.
So there is no `cached` events to handle as the prefix change.
Unless, the comment (code, i.e. set the state to pruning) at https://github.com/maidsafe/routing/pull/1798/files#r326167029 shall be carried out immediately when `casting the ParsecPrune` vote. However, this will be problematic I think?

Reply:
Things are still in the chain_accumulator, or even in parsec not yet consensused

Reply:
will have a double check